### ISSUE TITLE
No retries / queue / lost answers when SMTP server was not available

### ISSUE THREAD

**Issue by jacotec** [zammad/zammad#1239]
My SMTP server was temporarily not available. I just noticed by random when calling a ticket that outgoing mails during that time seems just not being sent out!

Inside the ticket which I opened is a yellow notification bar mentioning:

> Zustellung fehlgeschlagen: "Unable to send email to 'Max Muster <max.muster@gmail.com>': Can't use Channel::Driver::Smtp: #<Errno::ECONNREFUSED: Connection refused - connect(2) for "my.mailserver.de" port 587>"

Ah, and that's it???

I never would be aware of this if I had not opened the ticket. The history does not show that the answer was sent to the customer. And: I have no clue if there are more tickets in the system with similar issues.

Apart from some concerns this raises questions:

- What is the time frame and frequency Zammad retries to deliver mails when it was not successful doing so?

- Why didn't I get notifications/alarms that answers were not sent in Zammad, especially as Zammad seems to have given up in retrying to get them out?

- Where is the queue / list with outstanding answers / notifications which are not successfully sent?

- How do I retransmit any outstanding / not transmitted answers/notifications (also from other groups where I as the admin are not member of)?

I still hope that I made a mistake and are really overlooking something ... being too stupid or lacking sleep right now ;-))

If not, that's a major bug which causes messages to customers getting lost / not being sent out without any alarms/automatic retries.

EDIT: In "Monitoring" I see the entry "Aktueller Status: unprocessable mails 9". So there are more messages waiting for delivery ... mailserver is back up since 24 hours - how do I get these mails out?

**Comment by martini** [comment]
Hi @jacotec 

> Apart from some concerns this raises questions:

> What is the time frame and frequency Zammad retries to deliver mails when it was not successful doing so?

Zammad tries 4 times to deliver the mail via SMTP. The interval is try_count * 20 seconds. 

See also: https://github.com/zammad/zammad/blob/develop/app/models/observer/ticket/article/communicate_email/background_job.rb#L163

> Why didn't I get notifications/alarms that answers were not sent in Zammad, especially as Zammad seems to have given up in retrying to get them out?

There is already an issue for that https://github.com/zammad/zammad/issues/1198 @rolfschmidt is implementing it right now.

> Where is the queue / list with outstanding answers / notifications which are not successfully sent?

It's done via the background jobs. Zammad currently is using delayed jobs. To see all your open background jobs you can do it via rails console:

```
zammad run rails c
Delayed::Job.count
Delayed::Job.all
```

> How do I retransmit any outstanding / not transmitted answers/notifications (also from other groups where I as the admin are not member of)?

Basically it's done automatically (via delayed jobs). Usually in this cases the recipient is not correct, so the agent need to fix it manually by creating a new answer with correct recipient.

If you have questions, feel free.

 -Martin

**Comment by martini** [comment]
@jacotec just reopen on feedback of if something is not clear. Thanks a lot!

**Comment by jacotec** [comment]
Hi @martini 

Sorry for the delay, I was in vacancy enjoying some sun :-)

> Basically it's done automatically (via delayed jobs). Usually in this cases the recipient is not correct, so the agent need to fix it manually by creating a new answer with correct recipient.

In my case the SMTP server was not available during the retry time. I've recreated the answer and send it again (which means to copy/paste and to attach the attachment(s) ). Maybe a "Resend" button in case of a failure would to the job as a future expansion? ;-)

I've seen #1198 is implemented ... let's see how this works. But maybe think about the "Resend" link/button in the ticket in case of a final failure *g*

**Comment by Moulde** [comment]
I agree, a retry button would be pretty great in this case.

**Comment by MrGeneration** [comment]
@Moulde please don't recycle closed issues.
If you want to suggest features, I'd like to suggest opening a feature request topic on the Community if this topic doesn't exist yet (https://community.zammad.org).

If you require this functionality as a must have, you can also contact us at sales [at] zammad [dot] com for a custom development if needed.

To reduce noise on this repository I'll be limiting the conversation on this issue now. It's clsoed for 3 years now. This is nothing against you!
